pkg/apis/enterprise/v1/common_types.go

```diff
@@ -146,6 +146,12 @@ type VolumeSpec struct {
 
 	// Remote Storage type.
 	Type string `json:"storageType"`
+
+	// App Package Remote Store provider.
+	// For e.g. aws, azure, minio, etc.
+	// Currently we are only supporting aws.
+	// TODO: Support minio as well.
+	Provider string `json:"provider"`
```

Comment:
Nit: As discussed earlier, unless we really need to differentiate a particular client implementation, storage type(above) should be good enough to differentiate?

Reply:
Yes subba, we need to differentiate based on the provider. storageType alone won't be enough as we need provider specific SDK libraries at the minimum to make it work.